plutus-tx/src/PlutusTx/Foldable.hs

```diff
@@ -225,6 +228,20 @@ find p = coerce . foldMap (\ x -> First (if p x then Just x else Nothing))
 -- | Plutus Tx version of 'Data.Foldable.mapM_'.
 {-# INLINABLE mapM_ #-}
 mapM_ :: (Foldable t, Haskell.Monad m) => (a -> m b) -> t a -> m ()
+-- See Note [Haskell.Monad]
 mapM_ f = foldr c (Haskell.return ())
   where c x k = f x Haskell.>> k
         {-# INLINE c #-}
```

Comment:
What's even the point of having this function? It's a bad version of `traverse_`.